controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java

```diff
@@ -125,7 +123,8 @@ private void checkIntegration(Integration integration) {
     private void enqueue(WorkflowHandler handler, String integrationId) {
         executor.submit(() -> {
             Integration integration = dataManager.fetch(Integration.class, integrationId);
-            if (stale(handler, integration)) {
+            String scheduledKey = "" + integration.getDesiredStatus() + ":" + integrationId;
```

Comment:
Not 100% sure about the key. Shouldn't it be also the currentStatus included ? E.g. when the desiredStatus stays the same but the current status changes, shouldn't it be then rechecked, too ?

could be that i'm talking bs here ;-)